lib/pure/math.nim

```diff
@@ -1002,19 +1053,15 @@ func frexp*[T: float32|float64](x: T): tuple[frac: T, exp: int] {.inline.} =
       doAssert frexp(Inf).frac == Inf # +- Inf preserved
       doAssert frexp(NaN).frac.isNaN
 
-  when not defined(js):
-    var exp: cint
-    result.frac = c_frexp(x, exp)
-    result.exp = exp
-  else:
+  template frexpImpl: untyped =
     if x == 0.0:
       # reuse signbit implementation
-      let uintBuffer = toBitsImpl(x)
-      if (uintBuffer[1] shr 31) != 0:
+      let sign = x.signbit
+      if sign:
```

Comment:
```suggestion
      if x.signbit:
```

The temporary is unnecessary (I didn't see any other usage) and it's probably to keep these routines lighter of extraneous operations.